src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -228,12 +228,15 @@ JRT_LEAF(jlong, SharedRuntime::lrem(jlong y, jlong x))
 JRT_END
 
 
+#ifdef _WIN64
 const juint  float_sign_mask  = 0x7FFFFFFF;
 const juint  float_infinity   = 0x7F800000;
 const julong double_sign_mask = CONST64(0x7FFFFFFFFFFFFFFF);
 const julong double_infinity  = CONST64(0x7FF0000000000000);
+#endif
 
-JRT_LEAF(jfloat, SharedRuntime::frem(jfloat  x, jfloat  y))
+#if !defined(TARGET_COMPILER_gcc) || defined(_WIN64)
```

Comment:
That is the wrong condition - it would activate for any non-Windows, non-gcc compiler. This code is only for _WIN64 is it not? regardless of the (theoretical) compiler used on win64?

Reply:
For non-Windows non-gcc compiler it will use:
>   return ((jfloat)fmod((double)x,(double)y));

Which I find correct.

Reply:
Sorry my mistake.